cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorCustomCertTest.java

```diff
@@ -62,17 +66,91 @@
     private final String clusterName = "testkafka";
     protected static Vertx vertx;
 
-    @BeforeAll
-    public static void before() {
+    private Kafka kafka;
+    private KafkaCluster kafkaCluster;
+    private ResourceOperatorSupplier supplier;
+    private KafkaAssemblyOperator operator;
+    private ArgumentCaptor<StatefulSet> reconcileStsCaptor;
+    private ArgumentCaptor<Function<Pod, String>> isPodToRestartFunctionCaptor;
+
+    /**
+     * Mock the KafkaAssemblyOperator and override reconcile to only run through the steps we want to test
+     */
+    class MockKafkaAssemblyOperator extends KafkaAssemblyOperator  {
+        public MockKafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa, CertManager certManager, PasswordGenerator passwordGenerator, ResourceOperatorSupplier supplier, ClusterOperatorConfig config) {
+            super(vertx, pfa, certManager, passwordGenerator, supplier, config);
+        }
+
+        @Override
+        Future<Void> reconcile(ReconciliationState reconcileState)  {
+            return reconcileState.reconcileCas(this::dateSupplier)
+                    .compose(state -> state.getKafkaClusterDescription())
+                    .compose(state -> state.customTlsListenerCertificate())
+                    .compose(state -> state.customExternalListenerCertificate())
+                    .compose(state -> state.kafkaStatefulSet())
+                    .compose(state -> state.kafkaRollingUpdate())
+                    .map((Void) null);
+        }
+    }
+
+    @BeforeEach
+    public void before() {
```

Comment:
I think some of these will cause lot of conflicts with my changes in #2737. Some things such as the metrics do not work well when you are opening and closing vertx too often. So I actually mmoved most of these to @BeforeAll / @AfterAll and tied to keep the method names as `before` and `after`. Was there any particular reason why moving Vet.x to @BeforeEach / @AfterEach?

Reply:
You're absolutely right, I have changed it to follow your example.